clang/include/clang/Basic/LangOptions.def

```diff
@@ -233,6 +233,7 @@ LANGOPT(GPUMaxThreadsPerBlock, 32, 256, "default max threads per block for kerne
 LANGOPT(SYCLIsDevice      , 1, 0, "Generate code for SYCL device")
 LANGOPT(SYCLIsHost        , 1, 0, "SYCL host compilation")
 LANGOPT(SYCLAllowFuncPtr  , 1, 0, "Allow function pointers in SYCL device code")
+LANGOPT(SYCLNewKernelParamReq, 1, 0, "Enable trivially copyable requirement instead of standard layout requirement for SYCL kernel parameters")
```

Comment:
I disagree with this choice. Trvially copyable should always be required. Even before we fix the SYCL spec it is required by the C++ spec. This option should only  enable/disable whether std-layout is also checked. And the same way as our other usability improvements the default should be that the std-layout check is off.

Reply:
So, we will always require trivially copyable and enable/disable standard layout using option.
@Ruyk , What do you think about it?

Reply:
Yeah that makes some sense. The SYCL standard requires standard layout, which is why i would expect -sycl-std=sycl-1.2.1 to ask for standard layout.

Reply:
@Ruyk, do you have any preference how non-standard extensions should be controlled by users?
I've created an issue here: https://github.com/intel/llvm/issues/806 and I think it would be great to have a consistent approach across SYCL implementation.